lib/gui/os/dialog/services/dialog.js

```diff
@@ -113,20 +113,28 @@ module.exports = function($q, SupportedFormatsModel) {
    * });
    */
   this.showWarning = (options) => {
+    const BUTTONS = [
+      options.confirmationLabel,
+      options.rejectionLabel
+    ];
+
+    const BUTTON_CONFIRMATION_INDEX = _.indexOf(BUTTONS, options.confirmationLabel);
+    const BUTTON_REJECTION_INDEX = _.indexOf(BUTTONS, options.rejectionLabel);
```

Comment:
Given that we've _only_ got two buttons here, could you just do:
```js
const BUTTON_CONFIRMATION_INDEX = 0;
const BUTTON_REJECTION_INDEX = 1;
```
rather than using `indexOf`? As you say below, this would currently go wrong if `confirmationLabel` and `rejectionLabel` were the same.